File: ssm-sdk-client/src/main/java/ssm/client/json/JSONConverter.java
package ssm.client.json;

import okhttp3.ResponseBody;

import java.util.List;
import java.util.Optional;
import java.util.function.Function;

public interface JSONConverter {

    <T> Function<ResponseBody, List<T>> toCompletableObjects(Class<T> clazz);
    <T> Function<ResponseBody, Optional<T>> toCompletableObject(Class<T> clazz);

}

File: ssm-sdk/ssm-sdk-json/src/main/kotlin/ssm/sdk/json/JSONConverter.kt
package ssm.sdk.json

interface JSONConverter {
	fun <T> toCompletableObjects(clazz: Class<T>, value: String): List<T>
	fun <T> toCompletableObject(clazz: Class<T>, value: String): T?
	fun <T> toObject(clazz: Class<T>, value: String): T?
}
